client/state/reader/follows/reducer.js

```diff
@@ -20,21 +22,34 @@ import {
  * @param  {Object} action Action payload
  * @return {Array}        Updated state
  */
-export function items( state = [], action ) {
-	switch ( action.type ) {
-		case READER_FOLLOW:
-			return union( state, [ action.url ] );
-
-		case READER_UNFOLLOW:
-			return without( state, action.url );
-
-		case SERIALIZE:
-		case DESERIALIZE:
-			return [];
-	}
-
-	return state;
-}
+export const items = createReducer( {}, {
+	[ READER_FOLLOW ]: ( state, { url } ) => {
+		const urlKey = prepareComparableUrl( url );
+		return {
+			...state,
+			[ urlKey ]: { is_following: true },
+		};
+	},
+	[ READER_UNFOLLOW ]: ( state, { url } ) => {
+		const urlKey = prepareComparableUrl( url );
+		return {
+			...state,
+			[ urlKey ]: { is_following: false },
+		};
+	},
+	[ READER_FOLLOWS_RECEIVE ]: ( state, { follows } ) => {
+		const keyedNewFollows = reduce( follows, ( hash, follow ) => {
```

Comment:
FFTI:

Maybe you could use `_.keyBy`?

```javascript
const normalizedFollows = _.map( follows, follow => {
   ...follow,
   URL: prepareComparableUrl( follow.URL );,
   is_following: true,
} );
return merge( {}, state, keyBy( normalizedFollows, 'URL') )
```

I think your current implementation is more efficient